### ISSUE TITLE
Merge this branch with Keras 2.0

### ISSUE THREAD

**Issue by rravu3** [dmlc/keras#66]
Hi,

Wanted to ask what are the plans to pull in Keras 2.0 as there have been interface level changes from 1.2.2.

Rahul

**Comment by kevinthesun** [comment]
Yes. We will first finalize keras/mxnet for 1.2.2 and then switch to 2.0.

**Comment by yliu7366** [comment]
Waiting for MXNet + Keras 2. I have networks developed based on Keras 2 and I am desperate for MXNet auto multi-GPU support....

**Comment by TimZaman** [comment]
@kevinthesun timeline? Why not try to merge this into Keras?

**Comment by edmondja** [comment]
:'( same as @yliu7366 I hope at least somebody has begun to work on this

**Comment by davideboschetto** [comment]
It would be great to have an update: are things still planned for a Keras2.0 upgrade?

**Comment by kevinthesun** [comment]
@sandeep-krishnamurthy

**Comment by sandeep-krishnamurthy** [comment]
Hi Yes. This is planned and work is in progress. I have implemented all basic tensor operators.. https://github.com/simpledl/keras/tree/mxnet_backend_keras2

**Comment by davideboschetto** [comment]
This is great @sandeep-krishnamurthy !
Keep up the good work, having the backend in keras2 would be amazing, thanks!
Do you by any chance have a ballpark estimation for when the PR will come?

**Comment by sandeep-krishnamurthy** [comment]
Hello,
Keras2+MXNet is almost complete. https://github.com/deep-learning-tools/keras/tree/keras2_mxnet_backend . Please try and provide your feedback.

I have created a PR - https://github.com/keras-team/keras/pull/9291
However, there are few design incompatibility with MXNet and Keras in the way Model compile and Optimizers are handled. Hence, few changes are requested by Keras community. These changes require significant modifications in MXNet Module and KVStore components. Hence, we talked to keras community to release the fork for the benefit of Keras users as the interface remains the same.

**Comment by edmondja** [comment]
@sandeep-krishnamurthy How to install and try the Keras2-compatible version ?

**Comment by sandeep-krishnamurthy** [comment]
@edmondja - Thanks for you interest.
Please open issue at -  https://github.com/deep-learning-tools/keras if you face any problems.
```
$ mkdir keras
$ git clone https://github.com/deep-learning-tools/keras ./keras/
$ cd keras/

# MXNet backend code is in keras2_mxnet_backend branch. Check it out.
$ git fetch
$ git checkout keras2_mxnet_backend
$ git pull

# Install Keras
$ sudo python setup.py install
```

**Comment by edmondja** [comment]
Hello, I used all of your instructions but I have this error :

  File "/Users/edmondjacoupeau/anaconda/lib/python3.6/site-packages/keras/backend/__init__.py", line 72, in <module>
    assert _backend in {'theano', 'tensorflow', 'cntk'}

AssertionError

**Comment by sandeep-krishnamurthy** [comment]
Hi @edmondja It looks like you have not checked out keras2_mxnet_backend branch. Can you please confirm?
Because, in init.py you should see assert_backend in {'theano', 'tensorflow', 'cntk', 'mxnet'} like this 
https://github.com/deep-learning-tools/keras/blob/keras2_mxnet_backend/keras/backend/__init__.py#L72

**Comment by sandeep-krishnamurthy** [comment]
@edmondja - Did you had pre-installed keras (without mxnet backend)? If yes, please uninstall it before running above provided command. 
```
$ pip uninstall keras
```

**Comment by edmondja** [comment]
Thanks it worked
